feat: update the Terraform provider to version 1.9.0 and refactor the input variables.

akocbek opened the pull request:
### Description

<!--- Replace this text with a summary of the changes in this PR. Include why the changes are needed and context about the changes. List required dependencies. If there is a Git issue for the change, please link to it. --->

refactored:
- `create_s2s_auth_policy` renamed to `skip_secrets_manager_iam_auth_policy`, default value changed to false.
-  removed `secrets_manager_id` input variable
- added  validation for `create_policy` input variable

### Release required?
<!--- Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning). --->

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content
Small improvements:
- terraform provider version increased to `1.9.0`
- `create_s2s_auth_policy` renamed to `skip_secrets_manager_iam_auth_policy`, default value is set to `false`.
- removed `secrets_manager_id` input variable
- added  validation for `create_policy` input variable

<!--- If a release is required, replace this text with information that users need to know about the release. Write the release notes to help users understand the changes, and include information about how to update from the previous version.

Your notes help the merger write the commit message for the PR that is published in the release notes for the module. --->

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

akocbek commented:
/run pipeline

rajatagarwal-ibm reviewed (approved):
I had a quick chat with @akocbek, and they informed me that both of the solutions are being phased out. As a result, there's no need to bump the Terraform versions in these solutions, since the new, fully configurable solution will come with the updated version.

This PR looks good to me.

akocbek commented:
refactored:
- `create_s2s_auth_policy` renamed to `skip_secrets_manager_iam_auth_policy`, default value changed to false.
-  removed `secrets_manager_id` input variable
- added  validation for `create_policy` input variable

akocbek commented:
/run pipeline

ocofaigh commented on variables.tf:
```diff
@@ -94,16 +99,10 @@ variable "access_group_name" {
-  type        = string
-  description = "ID of the Secrets Manager that contains the certificate to use for the VPN, only required when create_s2s_auth_policy is true."
-  default     = null
+  description = "Whether to create an IAM authorization policy that permits communication between all VPN Servers (scoped to the given resource group) and the given Secrets Manager instance. Currently not possible to scope the policy to the exact VPN server ID since the policy is needed before the instance exists as it uses the cert stored in secrets manager during the provisioning process."
```
should probably give more info on the exact role it grants instead of just saying "permits communication"

akocbek replied:
updated

ocofaigh commented on variables.tf:
```diff
@@ -80,6 +80,11 @@ variable "create_policy" {
   default     = true
+
+  validation {
+    condition     = !(var.create_policy == true && var.access_group_name == null)
```
Is this not easier to read?
`var.create_policy ? var.access_group_name != null : true`

ocofaigh commented on main.tf:
```diff
@@ -2,26 +2,21 @@ locals {
 # without the cert from secrets manager, but it cant grab the cert from secrets manager until the policy is created.
 resource "ibm_iam_authorization_policy" "policy" {
-  count                       = var.create_s2s_auth_policy ? 1 : 0
+  count                       = !var.skip_secrets_manager_iam_auth_policy ? 1 : 0
```
is this not easier to read?
`var.skip_secrets_manager_iam_auth_policy ? 0 : 1`

ocofaigh reviewed (changes requested):
Looking better - some minor comments

akocbek commented:
/run pipeline

terraform-ibm-modules-ops commented:
:tada: This PR is included in version 2.2.0 :tada:

The release is available on:
- [GitHub release](https://github.com/terraform-ibm-modules/terraform-ibm-client-to-site-vpn/releases/tag/v2.2.0)
- `v2.2.0`

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
